lib/posix/pthread.c

```diff
@@ -1073,6 +1073,49 @@ int pthread_atfork(void (*prepare)(void), void (*parent)(void), void (*child)(vo
 	return ENOSYS;
 }
 
+/* this should probably go into signal.c but we need access to the lock */
+int pthread_sigmask(int how, const sigset_t *ZRESTRICT set, sigset_t *ZRESTRICT oset)
+{
+	struct posix_thread *t;
+
+	if (!(how == SIG_BLOCK || how == SIG_SETMASK || how == SIG_UNBLOCK)) {
+		return EINVAL;
+	}
+
+	t = to_posix_thread(pthread_self());
+	if (t == NULL) {
+		return ESRCH;
+	}
+
+	K_SPINLOCK(&pthread_pool_lock) {
+		if (oset != NULL) {
+			*oset = t->sigset;
+		}
+
+		if (set == NULL) {
+			K_SPINLOCK_BREAK;
+		}
+
+		switch (how) {
+		case SIG_BLOCK:
+			for (size_t i = 0, N = ARRAY_SIZE(set->sig); i < N; ++i) {
```

Comment:
Wouldn't this be enough here i.e., why do we need `N` variable
```
for (size_t i = 0; i < ARRAY_SIZE(set->sig); ++i) {
```

Reply:
Yes, good catch